doc/user-guide.md

```diff
@@ -464,6 +465,49 @@ func main() {
 }
 ```
 
+##### If 3rd party resource does not have `AddToScheme()` function
+
+Use `AddKnownTypes()` and `AddToGroupVersion()` functions.
+
+Example of registering `DNSEndpoints` 3rd party resource from `external-dns`:
+
+```go
+import (
+  ...
+    "k8s.io/apimachinery/pkg/runtime/schema"
+  ...
+   // DNSEndoints
+   externaldns "github.com/kubernetes-incubator/external-dns/endpoint"
+   metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+ )
+
+func main() {
+  ....
+
+    log.Info("Registering Components.")
+
+   scheme := mgr.GetScheme()
+
+   // Add external-dns DNSEndpoints resource
+   externaldnsSchemaGroupVersion := schema.GroupVersion{Group: "externaldns.k8s.io", Version: "v1alpha1"}
+   scheme.AddKnownTypes(externaldnsSchemaGroupVersion, &externaldns.DNSEndpoint{})
+   scheme.AddKnownTypes(externaldnsSchemaGroupVersion, &externaldns.DNSEndpointList{})
+   // https://github.com/kubernetes/kubernetes/pull/57243/files
+   // without metav1 we will face `Failed to list *endpoint.DNSEndpoint: v1.ListOptions is not suitable for converting to "externaldns.k8s.io/v1alpha1" in scheme "k8s.io/client-go/kubernetes/scheme/register.go:65"`
+   metav1.AddToGroupVersion(scheme, externaldnsSchemaGroupVersion)
```

Comment:
If a 3rd party resource doesn't provide its own scheme builder then we can construct one easily via the controller-runtime's [SchemeBuilder](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/scheme/scheme.go#L70-L77) utility which would provide the AddToScheme() function.

```Go
import (
	"k8s.io/apimachinery/pkg/runtime/schema"
	"sigs.k8s.io/controller-runtime/pkg/scheme"
)
...

schemeBuilder = &scheme.Builder{GroupVersion: schema.GroupVersion{Group: "externaldns.k8s.io", Version: "v1alpha1"}}
schemeBuilder.Register(&externaldns.DNSEndpoint{}, &externaldns.DNSEndpointList{})

if err := schemeBuilder.AddToScheme(mgr.GetScheme()); err != nil {
    log.Error(err, "")
    os.Exit(1)
}	  
```

@ytsarev I think this is probably cleaner than directly adding the types and metav1 groupversion.

Reply:
@hasbro17 amazing, thanks a lot for feedback, exactly what I was looking for. I've tested  the code and amended PR.